### ISSUE TITLE
Repeated runs gives "DEBUG: Checksum too old for [file] ..." on same files

### ISSUE THREAD

**Issue by jim-collier** [sahib/rmlint#347]
**Relevant output**:

`DEBUG: Checksum too old for 20111002-190456_SonDSCR1_2556.jpg, 0 < 1317611096`

Repeated runs with "write xattr" flag set, gives same results on the same files.

**Command**:

`sudo rmlint "/mnt/btrfs/array-f7" --no-crossdev --no-followlinks --see-symlinks --hidden -o progressbar -o summary --no-with-color --xattr-write --write-unfinished --algorithm=blake2bp -vvv --merge-directories --xattr-read --rank-by=pHMLDa -o csv -o json -o sh -o py 2>&1 | tee "output.log"`

 I can run and log with gdb and valgrind if necessary.

**Comment by sahib** [comment]
Is it only happening on those files? What kind of file is ``20111002-190456_SonDSCR1_2556.jpg``, regular or symlink? Can you do a ``getfattr -d 20111002-190456_SonDSCR1_2556.jpg``?

By the way, having both ``--no-followlinks`` and ``--see-symlinks`` at the same time does not make any sense. Also note that ``--merge-directories`` will set ``--see-symlinks`` anyways. You can only overwrite that later if you do ``--no-followlinks`` after.

**Comment by sahib** [comment]
The reason for this message is really dumb: This log message was being print for every file that was checked for xattrs. If it did not have a xattr, you'd see this message. This should have been checked beforehand, but there was a typo that caused it to progress one branch further. The version in ``develop`` should have it fixed (which I recommend you for testing).

**Comment by jim-collier** [comment]
I recompiled with dev build but am still getting the same "DEBUG: Checksum too old" lines, on the same ~753k files.

I recompiled with:

`git clone -b develop https://github.com/sahib/rmlint.git`
`cd rmlint/`
`scons config`
`sudo scons DEBUG=1 GDB=1 --prefix=/usr install`

Version:

`/usr/bin/rmlint --version`
`version 2.8.0 compiled: Jan 26 2019 at [15:06:40] "Maidenly Moose" (rev 87d7f964)`
`compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support`

Command run:

`gdb -ex r --args /usr/bin/rmlint "/mnt/btrfs/array-f7" --types=none --no-crossdev --hidden --xattr-write --write-unfinished --algorithm=blake2bp -vvv -o progressbar -o summary --no-with-color --merge-directories --xattr-read --rank-by=pHMLDa -o csv -o json -o sh -o py`

**Comment by sahib** [comment]
> I recompiled with dev build but am still getting the same "DEBUG: Checksum too old" lines, on the same ~753k files.

Okay, that's pretty weird. Is it still ``Checksum too old .* 0 < $timestamp``? Does it include the full path? You seem to run the right version (87d7f964).

**Comment by sahib** [comment]
Aha. Are you 100% certain that you run the right binary then? Version in ``develop`` [outputs the full path](https://github.com/sahib/rmlint/blob/develop/lib/xattr.c#L271), that's why I asked.

**Comment by jim-collier** [comment]
Oh shoot, I had reverted to a previous one. Let me check the logged stderr
and stdout when I get a chance, from the previous run via latest dev
version (at the time) and reply back...

I need to add `rmlint - - version` to my logs...

**Comment by jim-collier** [comment]
> Aha. Are you 100% certain that you run the right binary then? Version in `develop` [outputs the full path](https://github.com/sahib/rmlint/blob/develop/lib/xattr.c#L271), that's why I asked.

OK I'm finally back at it. I compiled the latest develop build, with the following commands:

```
sudo apt-get remove rmlint  ## Just in case it's installed via apt
cd $(mktemp -d)
git clone https://github.com/sahib/rmlint.git --branch develop --single-branch develop
cd develop
scons config
sudo scons --prefix=/usr install
```

Executing `rmlint --version` yields:
```
version 2.8.0 compiled: Jan 27 2019 at [00:04:26] "Maidenly Moose" (rev b4f1a33c)
compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support

rmlint was written by Christopher <sahib> Pahl and Daniel <SeeSpotRun> Thomas.
The code at https://github.com/sahib/rmlint is licensed under the terms of the GPLv3.
```

But I'm still not seeing full paths in the debug output.

Here's a sample line:
```
DEBUG: Checksum too old for BCP divide poster.pdf, 0 < 1546729450
```

Thoughts?

**Comment by sahib** [comment]
> Thoughts?

Yes,  I don't think it's January on your system:

> version 2.8.0 compiled: **Jan 27 2019** at [00:04:26] "Maidenly Moose" (rev b4f1a33c)

The binary you're executing is definitely not the recent one. Do a ``which rmlint`` to check what you're executing (and delete that).

**Comment by jim-collier** [comment]
Huh, I must have assumed I couldn't make a mistake, and didn't think too hard about the meaning of that date. Or something. I updated my build script to reduce the odds of that happening again.

It appears now I'm using the actual latest develop build.

```
rmlint --version
version 2.8.0 compiled: Jun 10 2019 at [08:35:25] "Maidenly Moose" (rev 4b626f23)
compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support
```

And now I'm seeing the full paths. For example:
```
DEBUG: Checksum too old for /mnt/btrfs/array-f7/0-0/0_findhome/BIOS Update Readme.pdf, 1225564827 < 1225564827
```

When I run this command:
```
getfattr -d "/mnt/btrfs/array-f7/0-0/0_findhome/BIOS Update Readme.pdf"
```

I get:
```
# file: mnt/btrfs/array-f7/0-0/0_findhome/BIOS Update Readme.pdf
user.rmlint.blake2bp.cksum="24942f182c69354e1ae887fc8117e6d70ef890e23c13e776df19151279280a0517aa26e49ca234e504c20b4a619c5d4c1be0c7bf3a7a560f848978f217db744c"
user.rmlint.blake2bp.mtime="1225564827.000000000"
user.rmlint.sha256.cksum="5c5da27ec251653d60ecda90f7939aef8094681f58ca05e4f099f0d1735bebde"
user.rmlint.sha256.mtime="1225564827.000000000"
```

(The reason the sha256 hashes are there is becasue I tried that option first, before realizing that blake2 was not only longer, but faster. Several thousand files have sha256 hashes, but it seems that error only happens now on 13 of them.)

**Comment by SeeSpotRun** [comment]
I'm closing this because I assume it was fixed by #380

**Comment by SeeSpotRun** [comment]
Re-opening because this is confirmed as bug:
> I notice that the "mtime" of both sha256 and blake2bp checksums are the same. I'm pretty sure I didn't have them calculated at the same time (if that's even possible).
